client/ui/actions/addon-groups.js

```diff
@@ -0,0 +1,171 @@
+import { browserHistory } from 'react-router';
+
+import apiFetch from '../utils/apiFetch';
+
+
+// AddonGroups list
+export const FETCH_ADDON_GROUPS_REQUEST = 'FETCH_ADDON_GROUPS_REQUEST';
+export const FETCH_ADDON_GROUPS_SUCCESS = 'FETCH_ADDON_GROUPS_SUCCESS';
+export const FETCH_ADDON_GROUPS_FAILURE = 'FETCH_ADDON_GROUPS_FAILURE';
+export const RESET_ADDON_GROUPS = 'RESET_ADDON_GROUPS';
+
+// Create an addon group
+export const CREATE_ADDON_GROUP_REQUEST = 'CREATE_ADDON_GROUP_REQUEST';
+export const CREATE_ADDON_GROUP_SUCCESS = 'CREATE_ADDON_GROUP_SUCCESS';
+export const CREATE_ADDON_GROUP_FAILURE = 'CREATE_ADDON_GROUP_FAILURE';
+export const RESET_NEW_ADDON_GROUP = 'RESET_NEW_ADDON_GROUP';
+
+// Fetch a single addon group
+export const FETCH_ADDON_GROUP_REQUEST = 'FETCH_ADDON_GROUP_REQUEST';
+export const FETCH_ADDON_GROUP_SUCCESS = 'FETCH_ADDON_GROUP_SUCCESS';
+export const FETCH_ADDON_GROUP_FAILURE = 'FETCH_ADDON_GROUP_FAILURE';
+export const RESET_ADDON_GROUP = 'RESET_ADDON_GROUP';
+
+// Update an addon group
+export const UPDATE_ADDON_GROUP_REQUEST = 'UPDATE_ADDON_GROUP_REQUEST';
+export const UPDATE_ADDON_GROUP_SUCCESS = 'UPDATE_ADDON_GROUP_SUCCESS';
+export const UPDATE_ADDON_GROUP_FAILURE = 'UPDATE_ADDON_GROUP_FAILURE';
+export const RESET_UPDATE_ADDON_GROUP = 'RESET_UPDATE_ADDON_GROUP';
+
+
+function apiError(type, error) {
+  return {
+    type,
+    error,
+  };
+}
+
+function requestAddonGroups() {
+  return {
+    type: FETCH_ADDON_GROUPS_REQUEST,
+  };
+}
+
+function receivedAddonGroups(addonGroups) {
+  return {
+    type: FETCH_ADDON_GROUPS_SUCCESS,
+    addonGroups,
+  };
+}
+
+function requestCreateAddonGroup() {
+  return {
+    type: CREATE_ADDON_GROUP_REQUEST,
+  };
+}
+
+function receivedCreateAddonGroup(addonGroup, saveAndContinue) {
+  if (saveAndContinue) {
+    browserHistory.push(`/addon_groups/${addonGroup.id}/`);
+  } else {
+    browserHistory.push('/addon_groups/');
+  }
+
+  return {
+    type: CREATE_ADDON_GROUP_SUCCESS,
+    addonGroup,
+  };
+}
+
+function requestAddonGroup(pk) {
+  return {
+    type: FETCH_ADDON_GROUP_REQUEST,
+    pk,
+  };
+}
+
+function receivedAddonGroup(addonGroup) {
+  return {
+    type: FETCH_ADDON_GROUP_SUCCESS,
+    addonGroup,
+  };
+}
+
+function requestUpdateAddonGroup() {
+  return {
+    type: UPDATE_ADDON_GROUP_REQUEST,
+  };
+}
+
+function receivedUpdateAddonGroup(addonGroup, saveAndContinue) {
+  if (!saveAndContinue) {
+    browserHistory.push('/addon_groups/');
+  }
+
+  return {
+    type: UPDATE_ADDON_GROUP_SUCCESS,
+    addonGroup,
+  };
+}
+
+export function resetAddonGroup() {
+  return {
+    type: RESET_ADDON_GROUP,
+  };
+}
+
+export function resetCreateAddonGroup() {
+  return {
+    type: RESET_NEW_ADDON_GROUP,
+  };
+}
+
+export function resetUpdateAddonGroup() {
+  return {
+    type: RESET_UPDATE_ADDON_GROUP,
+  };
+}
+
+export function fetchAddonGroups() {
+  return dispatch => {
+    dispatch(requestAddonGroups());
```

Comment:
I think I'd make `requestAddonGroups` go away, and just put the object definition here, since it is only used in one place, and is very simple. I could go either way on this though.